File: JSPM.Web/Modules/Students/Academic/AcademicForm.cs
using Serenity.ComponentModel;

namespace JSPM.Students.Forms;

[FormScript("Students.Academic")]
[BasedOnRow(typeof(AcademicRow), CheckNames = true)]
public class AcademicForm
{
    public string CourseLevel { get; set; }
    public string CourseName { get; set; }
    public string YearOfPassing { get; set; }
    public string Percentage { get; set; }
    public string Remark { get; set; }
    public string PassingType { get; set; }
    public int ProjectId { get; set; }
    public int StudentId { get; set; }
    public int SkillId { get; set; }
}

File: JSPM.Web/Modules/ServerTypes/Students/AcademicForm.ts
import { StringEditor, IntegerEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface AcademicForm {
    CourseLevel: StringEditor;
    CourseName: StringEditor;
    YearOfPassing: StringEditor;
    Percentage: StringEditor;
    Remark: StringEditor;
    PassingType: StringEditor;
    ProjectId: IntegerEditor;
    StudentId: IntegerEditor;
    SkillId: IntegerEditor;
}

export class AcademicForm extends PrefixedContext {
    static formKey = 'Students.Academic';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!AcademicForm.init)  {
            AcademicForm.init = true;

            var w0 = StringEditor;
            var w1 = IntegerEditor;

            initFormType(AcademicForm, [
                'CourseLevel', w0,
                'CourseName', w0,
                'YearOfPassing', w0,
                'Percentage', w0,
                'Remark', w0,
                'PassingType', w0,
                'ProjectId', w1,
                'StudentId', w1,
                'SkillId', w1
            ]);
        }
    }
}